static/src/javascripts/projects/common/modules/commercial/commercial-features.js

```diff
@@ -1,38 +1,123 @@
+
+        var switches = config.switches;
+
+        // Feature switches
+
+        this.dfpAdvertising =
+            externalAdvertising &&
+            !sensitiveContent;
+
+        this.topBannerAd =
+            this.dfpAdvertising &&
+            !minuteArticle;
+
+        this.articleBodyAdverts =
+            this.dfpAdvertising &&
+            !minuteArticle &&
+            isArticle &&
+            !isLiveBlog &&
+            switches.standardAdverts;
+
+        this.articleAsideAdverts =
+            this.dfpAdvertising &&
+            !minuteArticle &&
+            !isMatchReport &&
+            !!(isArticle || isLiveBlog) &&
+            switches.standardAdverts;
+
+        this.sliceAdverts =
+            this.dfpAdvertising &&
+            !minuteArticle &&
+            switches.standardAdverts;
+
+        this.popularContentMPU =
+            this.dfpAdvertising &&
+            !minuteArticle;
+
+        this.videoPreRolls =
+            externalAdvertising &&
+            !sensitiveContent &&
+            switches.videoAdverts;
+
+        this.frontCommercialComponents =
+            this.dfpAdvertising &&
+            !minuteArticle &&
+            config.page.isFront &&
+            switches.commercialComponents;
+
+        this.thirdPartyTags =
+            externalAdvertising &&
+            !isIdentityPage;
+
+        this.badges =
+            externalAdvertising &&
+            switches.sponsored;
+
+        this.outbrain =
+            externalAdvertising &&
+            !sensitiveContent &&
+            switches.outbrain;
+
+        this.async = {
+            membershipMessages : detect.adblockInUse.then(function (adblockUsed) {
+                return !adblockUsed &&
+                    detect.getBreakpoint() !== 'mobile' &&
+                    isArticle &&
+                    !userFeatures.isPayingMember();
+            })
+        };
     }
 
-    /**
-     * Take the results of all our commercial content policies and smush them together
-     * @see commercial-feature-policies.js for the list.
-     */
-    function getFeatureSwitches() {
-        var activeSwitches = {};
-        var switchesForPolicies = commercialFeaturePolicies.getPolicySwitches();
-
-        forOwn(switchesForPolicies, function mergeSwitches(policySwitchSet) {
-            forOwn(policySwitchSet, function (switchValue, switchKey) {
-                var activeValue = activeSwitches[switchKey];
-                if (activeValue !== false) {
-                    // A policy can only override a true or undefined switch
-                    // This gives any policy the ability to veto a commercial feature
-                    activeSwitches[switchKey] = switchValue;
-                }
-            });
-        });
-
-        return activeSwitches;
+    try {
+        config.commercial = config.commercial || {};
+        return config.commercial.featuresDebug = new CommercialFeatures();
```

Comment:
I'd like to hint that production code shouldn't hang off this object - it should `require` in the service instead, which makes sure all the keys are set.